### PR TITLE
Adds mob bleeding overlays, tweaks some bleeding balance

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
This PR adds new mob overlays for humans that display when they are bleeding, which of their bodyparts are bleeding, and how much blood they're losing. Anyone who's gotten stabbed in game knows that losing blood can be extremely lethal when not treated quickly, and while #56056 was a good step in giving players feedback on if their own bleeding is getting better or worse, this makes it much easier to gauge bloodloss at a glance for not only yourself, but others as well.

[Here's a clip of the bleeding in action.](https://streamable.com/sxjpbc)

This is something I've wanted to do for a while, and now with sprite help from the always wonderful @MrDoomBringer, I've finally gotten around to adding it! The bleeding overlays are not hidden by clothing or equipment for visibility purposes, and the spurting and dripping animations are swapped out for static versions on dead/stasis'd people.

#### Misc bleeding changes
I realized there's some issues with bleeding and slash wounds that needed to be fixed in this PR. Due to an issue with how blood_flow was handled when a higher severity slash was demoted to a lower severity, one suture was enough to fully cure a critical slash wound. This has been fixed.

I also forgot I set the max bloodflow for a slash to 4, which is below the initial flow for a critical slash. I've raised the maximum flow to 4.5, so avulsions get worse over time as they're supposed to. To compensate, I lowered their initial flow, the rate they get worse at, and halved how much slashing damage increases bleeding on existing wounds. Grabbing yourself to stifle bloodloss is also quicker and slows you down less.

## Why It's Good For The Game
Makes bleeding much more visible, allowing players to quickly tell how badly someone needs medical attention before they drop dead of bloodloss. The fixes also make bleeding more consistent, so slash wounds no longer seem to disappear randomly.

## Changelog

:cl: Ryll/Shaps and MrDoomBringer
imageadd: Added new bleeding overlays for humans! People who are losing blood will now have overlays displayed on them that show which bodyparts are bleeding, and how much blood they're losing from that bodypart.
fix: Slash wounds now properly retain their bleed rate when being demoted to lower severities, meaning critical slashes will no longer be fully healed after using one suture.
balance: Critical slash wounds have had their initial bloodflow reduced, and existing slash wounds gain less bloodloss when hit with further slash attacks. The cap for how much blood a slash can lose per tick has been raised slightly.
balance: Grabbing your own bleeding limb to staunch the bleeding is quicker and slows you down less
/:cl:

### CODE REVIEW

File: code/__DEFINES/mobs.dm [hunk 1]
```diff
@@ -511,65 +511,67 @@
...
+#define BACK_LAYER 13
 /// Hair layer (mess with the fro and you got to go!)
-#define HAIR_LAYER 11 //TODO: make part of head layer?
+#define HAIR_LAYER 12 //TODO: make part of head layer?
 /// Facemask layer (gas masks, breath masks, etc.)
-#define FACEMASK_LAYER 10
+#define FACEMASK_LAYER 11
 /// Head layer (hats, helmets, etc.)
-#define HEAD_LAYER 9
+#define HEAD_LAYER 10
```
Comment (Mothblocks) [inline code/__DEFINES/mobs.dm:560]: If you want to save yourself a lot of future headache, change these to just be `HANDCUFF_LAYER + 1` etc, so only a few need to be changed.

File: code/__DEFINES/wounds.dm [hunk 2]
```diff
@@ -77,10 +77,12 @@ GLOBAL_LIST_INIT(global_all_wound_types, list(/datum/wound/blunt/critical, /datu
 // ~random wound balance defines
 /// how quickly sanitization removes infestation and decays per second
 #define WOUND_BURN_SANITIZATION_RATE 0.075
-/// how much blood you can lose per second per slash max. 4 is a LOT of blood for one cut so don't worry about hitting it easily
-#define WOUND_SLASH_MAX_BLOODFLOW 4
+/// how much blood you can lose per tick per slash max.
```
Comment (Mothblocks) [inline code/__DEFINES/wounds.dm:80]: Why per tick? Per second is always easier to think about

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 3]
```diff
@@ -982,20 +989,61 @@
...
 		QDEL_NULL(grasped_by)
 
 	return bleed_rate
 
+// how much blood the limb needs to be losing per tick (not counting laying down/self grasping modifiers) to get the different bleed icons
+#define BLEED_OVERLAY_LOW 0.5
+#define BLEED_OVERLAY_MED 1.5
+#define BLEED_OVERLAY_GUSH 3.25
+
+/obj/item/bodypart/proc/update_part_wound_overlay()
```
Comment (Mothblocks) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1011]: Can you make the other paths explicitly return FALSE? Unexpected FALSE over null causes weird gotchas, and is just a weird signature.

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 4]
```diff
@@ -982,20 +989,61 @@
...
+			bleed_rate *= 0.75
+		if(grasped_by)
+			bleed_rate *= 0.7
 
 	if(!bleed_rate)
 		QDEL_NULL(grasped_by)
 
 	return bleed_rate
 
+// how much blood the limb needs to be losing per tick (not counting laying down/self grasping modifiers) to get the different bleed icons
```
Comment (Mothblocks) [inline code/modules/surgery/bodyparts/_bodyparts.dm:1006]: Why tick?

File: code/modules/mob/living/living.dm [hunk 5]
```diff
@@ -859,6 +859,9 @@
 /mob/living/proc/update_damage_overlays()
 	return
 
+/mob/living/proc/update_wound_overlays()
```
Comment (Mothblocks) [inline code/modules/mob/living/living.dm:862]: Document

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Mothblocks** [comment]
looks spiffy

File: code/__DEFINES/wounds.dm (hunk 2 above)
Comment (Ryll-Ryll) [reply code/__DEFINES/wounds.dm:80]: The proc only runs every tick, but I spoke to you regarding the per second/per tick thing on discord

File: code/modules/surgery/bodyparts/_bodyparts.dm (hunk 4 above)
Comment (Ryll-Ryll) [reply code/modules/surgery/bodyparts/_bodyparts.dm:1006]: The proc only runs every tick, but I spoke to you regarding the per second/per tick thing on discord

**Pull request merged**